force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -124,13 +156,23 @@ public with sharing class DummyJSONCallout {
 	@TestVisible // Allows test class to see this method. Since it is private, it would not be visible otherwise.
 	private static String generateDummyJsonUserPayload(String contactId) {
 		// Query the contact to get the field values to generate the JSON payload.
-
+		Contact cont = [
+			SELECT Id, FirstName, LastName, Email, Phone, DummyJSON_Id__c
+			FROM Contact
+			WHERE Id = :contactId
+		];
 		// Create a map of the field values.
-
+		Map<String, Object> valuesMap = new Map<String, Object>();
+		valuesMap.put('salesforceId', contactId);
+		valuesMap.put('firstName', String.isNotBlank(cont.FirstName) ? cont.FirstName : 'unknown');
+		valuesMap.put('lastName', String.isNotBlank(cont.LastName) ? cont.LastName : 'unknown');
+		valuesMap.put('email', String.isNotBlank(cont.Email) ? cont.Email : 'unknown');
+		valuesMap.put('phone', String.isNotBlank(cont.Phone) ? cont.Phone : 'unknown');
```

Comment:
Nice work catching these values if they are empty.
Putting unknown for the email may cause an error.